include/core/reactor.h

```diff
@@ -96,191 +60,13 @@ void _lf_set_present(lf_port_base_t* port);
-    out->length = len; \
-} while(0)
-#endif
-/**
- * Version of set() for output types given as 'type[number]'.
- *
- * This sets the _is_present variable corresponding to the specified output
- * to true (which causes the array message to be sent). The values in the
- * output are normally written directly to the array or struct before or
- * after this is called.
- * @param out The output port (by name).
- */
-#define lf_set_present(out) \
-do { \
-   _lf_set_present((lf_port_base_t*)out); \
-} while(0)
+/// \endcond // INTERNAL
 
-/**
- * Version of set() for output types given as 'type*' or 'type[]' where you want
- * to forward an input or action without copying it.
- *
- * The deallocation of memory is delegated to downstream reactors, which
- * automatically deallocate when the reference count drops to zero.
- * @param out The output port (by name).
- * @param token A pointer to token obtained from an input or action.
- */
-#ifndef __cplusplus
-#define _LF_SET_TOKEN(out, newtoken) \
-do { \
-   _lf_set_present((lf_port_base_t*)out); \
-    _lf_replace_template_token((token_template_t*)out, newtoken); \
-    out->value = newtoken->value; \
-    out->length = newtoken->length; \
-} while(0)
-#else
-#define _LF_SET_TOKEN(out, newtoken) \
-do { \
-   _lf_set_present((lf_port_base_t*)out); \
-    _lf_replace_template_token((token_template_t*)out, newtoken); \
-    out->value = static_cast<decltype(out->value)>(newtoken->value); \
-    out->length = newtoken->length; \
-} while(0)
-#endif
-
-/**
- * Set the destructor used to free "token->value" set on "out".
- * That memory will be automatically freed once all downstream
- * reactions no longer need the value.
- *
- * @param out The output port (by name) or input of a contained
- *            reactor in form input_name.port_name.
- * @param destruct A pointer to a void function that takes a pointer argument
- *             or NULL to use the default void free(void*) function.
- */
-#define _LF_SET_DESTRUCTOR(out, destruct) \
-do { \
-    ((token_type_t*)out)->destructor = destruct; \
-} while(0)
-
-/**
- * Set the constructor used to copy construct "token->value" received
- * by a downstream mutable input.
- *
- * @param out The output port (by name) or input of a contained
- *            reactor in form input_name.port_name.
- * @param constructor A pointer to a void* function that takes a pointer argument
- *                 or NULL to use the memcpy operator.
- */
-#define _LF_SET_COPY_CONSTRUCTOR(out, constructor) \
-do { \
-    ((token_type_t*)out)->copy_constructor = constructor; \
-} while(0)
+//////////////////////  Macros for reading and writing ports  //////////////////////
+// NOTE: Ports passed to these macros can be cast to:
+// lf_port_base_t: which has the field bool is_present (and more);
+// token_template_t: which has a lf_token_t* token field; or
+// token_type_t: Which has element_size, destructor, and copy_constructor fields.
```

Comment:
This comment seem out-of-place?

Reply:
I'm working on another branch `further-cleanup` that will include significant cleaning of reactor.h and reactor_common.h.  Let's leave this for that PR.